src/librustc_typeck/check/_match.rs

```diff
@@ -38,12 +38,11 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 self.write_ty(pat.id, expected);
             }
             PatKind::Lit(ref lt) => {
-                self.check_expr(&lt);
-                let expr_ty = self.expr_ty(&lt);
+                let expr_t = self.check_expr(&lt);
```

Comment:
Because having a method named `expr_ty` and a variable named `expr_ty` is confusing.